File: 94. Binary Tree Inorder Traversal/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        res, tmp = [], []
        while root or len(tmp)>0:
            while root:
                tmp.append(root)
                root=root.left
            res.append(tmp[-1].val)
            root=tmp[-1].right
            del tmp[-1]
        return res

File: 94. Binary Tree Inorder Traversal/Solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    void inOrder(TreeNode* root, vector<int>& res) {
        if(root->left)
            inOrder(root->left,res);
        res.push_back(root->val);
        if(root->right)
            inOrder(root->right,res);
    }
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int>res;
        if(!root)
            return res;
        inOrder(root, res);
        return res;
    }
};